### PR TITLE
added chamferRect in runtimeobjects

### PR DESCRIPTION
**Pull request by trp02**
Gave issue #5741 a shot and added chamfer rectangle functions. 
Was wondering if I did correctly/on the right track. If yes then i'll try to add other shapes 
Feedback appreciated

### CODE REVIEW

File: Extensions/PrimitiveDrawing/shapepainterruntimeobject-pixi-renderer.ts [hunk 1]
```diff
@@ -136,6 +136,31 @@ namespace gdjs {
...
+        this._object._fillColor,
+        this._object._fillOpacity / 255
+      );
+      //@ts-ignore from @pixi/graphics-extras
+      this._graphics.drawChamferRect(
+        x1, 
+        y1, 
+        x2-x1,
+        y2-y1,
+        chamfer ? chamfer : 1
```
Comment (AlexandreSi) [inline Extensions/PrimitiveDrawing/shapepainterruntimeobject-pixi-renderer.ts:157]: I checked the code in PIXI (https://api.pixijs.io/@pixi/graphics-extras/src/drawChamferRect.ts.html) and you can see that if `chamfer <= 0`, a rectangle is drawn. So we could allow chamfer to take any value here.

**Review by AlexandreSi — COMMENTED** [review]
I think you've taken the right path with this PR. With these changes, the feature is now available with JS events (you made it available to ShapePainterRuntimeObject).
Now you will need to expose it to the editor through the extension declaration.

**Comment by trp02** [comment]
I addedthe  action to get object in JsExtension.cpp but I had some questions.

Why are some functions(drawStar, drawRoundedRect, etc) in their twice?

Also assuming the action needs to be added in [Extension](https://github.com/trp02/GDevelop/blob/chamferRect/Extensions/PrimitiveDrawing/Extension.cpp).cpp as well but I'm unsure as to what png's its referring to. Should I just not worry about write it following the structure?

**Comment by AlexandreSi** [comment]
Regarding the duplicates, it might be only duplicates, good catch! I think you can remove them in your PR.

Regarding Extension.cpp, you can follow the same structure. If you can do it, please add the needed images (you can look at `roundedRectangle.png` to see what is expected) so that a user understands what the action does with a glance.

**Comment by AlexandreSi** [comment]
Thanks for the pngs! Is it possible to have the corners transparent and only the filling in white?

**Comment by trp02** [comment]
Is there any more changes/additions needed for chamfer rectangles?
I can add the other shapes after I complete the chamfer.

**Comment by AlexandreSi** [comment]
I tried it out, it looks good to me!

**Comment by AlexandreSi** [comment]
The CI did not complete though, you will have to format your code: run `npm run format` in GDJS folder.

File: GDJS/package.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
     "minimist": "^1.2.5",
     "patch-package": "^6.4.7",
     "pixi.js": "7.3.0",
-    "prettier": "2.1.2",
+    "prettier": "^2.1.2",
```
Comment (AlexandreSi) [inline GDJS/package.json:19]: Did you really need this? We prefer having fixed dependencies to avoid using uncontrolled versions

Comment (trp02) [reply GDJS/package.json:19]: Probably not. I thought this change was a part of npm run format but I guess not. 
My git extension on vscode  is having issues and messing up changes/updates

**Comment by trp02** [comment]
I reverted the package changes. 
package-lock is showing a "diff" but it looks the same to me

**Review by AlexandreSi — APPROVED** [review]
LGTM

**Comment by AlexandreSi** [comment]
Thanks a lot @trp02 for this first PR to the GDevelop repository! 🙇
As you suggested, you can give a try to the 2 other shapes if you want.

**Pull request merged**